pkg/frontend/load.go

```diff
@@ -1023,17 +1129,29 @@ func rowToColumnAndSaveToStorage(handler *WriteBatchHandler, forceConvert bool,
 						nulls.Add(vec.Nsp, uint64(i))
 					} else {
 						field := line[j]
-						d, err := strconv.ParseFloat(field, 64)
-						if err != nil || d < math.MinInt8 || d > math.MaxInt8 {
-							logutil.Errorf("parse field[%v] err:%v", field, err)
-							if !ignoreFieldError {
-								return err
+						if judgeInterge(field) {
+							d, err := strconv.ParseInt(field, 10, 8)
```

Comment:
1, in float32 branch, ParseInt uses 8 bit is not enough.
2, Actually, we can use the original logic here. We want float32, we donot need the check it is integer anymore here.  It is same as following.